pkg/api/validation/validation.go

```diff
@@ -2444,6 +2421,8 @@ func validatePodAffinityTerm(podAffinityTerm api.PodAffinityTerm, allowEmptyTopo
 	}
 	if len(podAffinityTerm.TopologyKey) != 0 {
 		allErrs = append(allErrs, unversionedvalidation.ValidateLabelName(podAffinityTerm.TopologyKey, fldPath.Child("topologyKey"))...)
+	} else {
```

Comment:
Also, how does this break Pod anti-affinity?  You need a `topologyKey: kubernetes.io/hostname` for it to be valid.